keras_cv/datasets/pascal_voc/segmentation.py

```diff
@@ -0,0 +1,320 @@
+        os.path.join(data_dir, "JPEGImages", i + ".jpg") for i in image_ids
+    ]
+    p = multiprocessing.Pool(10)
+    metadata = p.map(parse_single_image, image_file_paths)
+
+    # Transpose the metadata which convert from list of dict to dict of list.
+    keys = [
+        "image/filename",
+        "image/file_path",
+        "segmentation/class/file_path",
+        "segmentation/object/file_path",
+        "labels",
+        "width",
+        "height",
+    ]
+    result = {}
+    for key in keys:
+        values = [value[key] for value in metadata]
+        result[key] = values
+
+    # The ragged objects need some special handling
+    for key in ["label", "pose", "bbox", "is_truncated", "is_difficult"]:
+        values = []
+        objects = [value["objects"] for value in metadata]
+        for object in objects:
+            values.append([o[key] for o in object])
+        result["objects/" + key] = values
+    return result
+
+
+def _load_images(example):
+    image_file_path = example.pop("image/file_path")
+    segmentation_class_file_path = example.pop("segmentation/class/file_path")
+    segmentation_object_file_path = example.pop("segmentation/object/file_path")
+    image = tf.io.read_file(image_file_path)
+    image = tf.image.decode_jpeg(image)
+
+    segmentation_class_mask = tf.io.read_file(segmentation_class_file_path)
+    segmentation_class_mask = tf.image.decode_png(segmentation_class_mask)
+
+    segmentation_object_mask = tf.io.read_file(segmentation_object_file_path)
+    segmentation_object_mask = tf.image.decode_png(segmentation_object_mask)
+
+    example.update(
+        {
+            "image": image,
+            "class_segmentation": segmentation_class_mask,
+            "object_segmentation": segmentation_object_mask,
+        }
+    )
+    return example
+
+
+def _build_dataset_from_metadata(metadata):
+    # The objects need some manual conversion to ragged tensor.
+    metadata["labels"] = tf.ragged.constant(metadata["labels"])
+    metadata["objects/label"] = tf.ragged.constant(metadata["objects/label"])
+    metadata["objects/pose"] = tf.ragged.constant(metadata["objects/pose"])
+    metadata["objects/is_truncated"] = tf.ragged.constant(
+        metadata["objects/is_truncated"]
+    )
+    metadata["objects/is_difficult"] = tf.ragged.constant(
+        metadata["objects/is_difficult"]
+    )
+    metadata["objects/bbox"] = tf.ragged.constant(
+        metadata["objects/bbox"], ragged_rank=1
+    )
+
+    dataset = tf.data.Dataset.from_tensor_slices(metadata)
+    dataset = dataset.map(_load_images, num_parallel_calls=tf.data.AUTOTUNE)
+    return dataset
+
+
+def load(
+    split="train",
+    data_dir=None,
+):
+    """Load the Pacal VOC 2012 dataset.
+
+    This function will download the data tar file from remote if needed, and untar to
+    the local `data_dir`, and build dataset from it.
+
+    Args:
+        split: string, can be 'train', 'eval', or None. When None, both train and eval data
```

Comment:
TFDS defaults to `None` for split, and IMO that experience is a bit more intuitive. It feels weird to have to pass split=None to get both splits

Reply:
For the keras fit/eval workflow, I think default to train actually make more sense, since the fit() method doesn't need eval data, or the eval data are actually specified by a separate param (validation_data).